lib/recent.go

```diff
@@ -0,0 +1,119 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	OpenTofu  []string `json:"openTofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, distribution string) {
+	if !validVersionFormat(requestedVersion) {
```

Comment:
I'm not sure I really understand the need for this check.
Since it's an internal method, it can't be called externally.
Do we really expect to have code paths that would allow an invalid version to be passed in, especially since it's called post-install

Reply:
I just added the check again as it already was in the old function that was adding versions to the recent file.